refactor: migrate order history to my account

sethidden opened the pull request:
This already has lang="ts" so there's no need to do anything
- [x] split order history into two routes: order list and single order details

atagisow-alokai committed:
refactor: move order history for my account

refactor: split order history into list and single file page

refactor: fix stale async fetch key

before this commit visiting any singleorder for any order number would
always show the first fetched order, despite the orderId changing

refactor: remove unused

refactor: remove getters

also misc changes: move properties to top since they appear first in
the template aswell

sethidden commented on packages/theme/modules/customer/composables/useUserOrder/useUserOrder.ts:
```diff
@@ -25,7 +25,7 @@ export interface UseUserOrderInterface {
   /**
    * Fetches orders of the current customer
    */
-  search(params: UseUserOrderSearchParams) : Promise<Maybe<Array<CustomerOrdersQuery['customer']['orders']['items']>>>;
```
not sure why was this taking types from the QUERY (the stuff you type in in eg. Altair), not the actual data returned

sethidden replied:
also fixed one ts-ignore that existed because of this issue

sethidden commented on packages/theme/modules/customer/pages/MyAccount/OrderHistory/OrderHistory.vue:
```diff
@@ -100,13 +50,12 @@
               <span :class="getStatusTextClass(order)">{{ orderGetters.getStatus(order) }}</span>
             </SfTableData>
             <SfTableData class="orders__view orders__element--right">
-              <SfButton
```
Changed button to link for some a11y gains

sethidden commented on packages/theme/plugins/fcPlugin.ts:
```diff
@@ -1,11 +1,19 @@
-import formatCurrency from '~/helpers/formatCurrency';
 import { Plugin } from '@nuxt/types';
+import formatCurrency from '~/helpers/formatCurrency';
```
Needed to use $fc in setup() and realized that context.$fc works fine, but only TypeScript didn't know that it exists on the context (see fcPlugin.js)

bartoszherba commented on packages/theme/modules/customer/pages/MyAccount/OrderHistory/OrderHistory.vue:
```diff
@@ -1,62 +1,12 @@
+  <SfTabs>
     <SfTab
       data-cy="order-history-tab_my-orders"
       :title="$t('My orders')"
```
![Screenshot 2022-05-11 at 15 13 25](https://user-images.githubusercontent.com/16045377/167858465-2a54a05c-1b0e-4236-82da-12e54982947b.png)

bartoszherba replied:
Could you check if we can avoid this problem?

sethidden replied:
This issue frequently occurs in `<script>` where you have to write eg. somePropThatTypeScriptExpectsToBeAString: context.app.i18n.t('some label') **as string**

The problem is I can't really use "as string" in the template. I could move it to the script, set as string here then make it available on the instance and use it like `:title="myLabel"` but that's kinda painful so I don't want to do this.

But props to JetBrains for the sophisticated implementation of the language server

bartoszherba commented on packages/theme/modules/customer/pages/MyAccount/OrderHistory/SingleOrder.vue:
```diff
@@ -0,0 +1,201 @@
+          All Orders
+        </SfButton>
+        <div class="highlighted highlighted--total">
+          <!-- TODO move to computed and use vfor -->
```
Is this todo still valid?

sethidden replied:
Removed, thanks

bartoszherba commented on packages/theme/modules/customer/pages/MyAccount/OrderHistory/SingleOrder.vue:
```diff
@@ -0,0 +1,201 @@
+
+    const currentOrder = computed(() => {
+      const {
+        getItems, getItemName, getItemQty, getItemPrice, getId, getDate, getStatus, getPrice, getItemSku,
```
I think that most (if not all) can be easily replaced. Easy and fast win to get rid on getters in this component

sethidden replied:
removed all references to getters in this component, thanks

bartoszherba reviewed (approved):
No critical issues but left few comments.

atagisow-alokai committed:
fix: remove randomness from get order id

atagisow-alokai committed:
fix: replace deprecated created_at with order_date

the orderGetters in the theme already try to get order_date property but fail because
the api-client middleware doesn't return that in the response payload

the created_at property is deprecated in magento

sethidden commented on packages/api-client/src/api/customerOrders/customerOrders.ts:
```diff
@@ -7,7 +7,7 @@ export default gql`
         items {
           order_number
           id
-          created_at
```
fyi @bartoszherba - added this aswell since your last review - getters try to get order_date but it's not returned from the api-client middleware. created_at was returned but it's deprecated in the graphql schema by adobe/magento

bartoszherba replied:
great!

sethidden commented on packages/theme/modules/checkout/getters/orderGetters.ts:
```diff
@@ -3,7 +3,7 @@ import { CustomerOrder, OrderItemInterface } from '~/modules/GraphQL/types';
 
 export const getDate = (order: CustomerOrder): string => new Date(order?.order_date).toLocaleDateString() || '';
 
-export const getId = (order: CustomerOrder): string => String(Number.parseInt(order?.number, 10) || Math.floor(Math.random() * 100));
```
@bartoszherba also removed this funny thing

bartoszherba replied:
k
